### DISCUSSION TITLE
Why function for applying query returns void instead of a promise?

### DISCUSSION THREAD

**Discussion by illyakurochkin** [dotansimha/graphql-code-generator#5520] [Ideas] [answered]
I'm using these plugins:
```
- typescript
- typescript-operations
- typescript-react-apollo
```

And when I generate hooks from operations I receive something like this.

```
const [fetch, {data, loading}] = useGetSomethingLazyQuery();
...
await fetch(); // I can't do that because `fetch` return type is void
```

Why can't it return a promise, like for mutations?

**Comment by ardatan** [comment, marked as answer]
React Apollo plugin is just a wrapper on top Apollo's own implementation. Codegen adds nothing special to their own `useLazyQuery` function. So it might be better to ask this on React Apollo's repo or Apollo's own community channel.

**Comment by pepijnolivier** [comment]
Upgrade to Apollo 3.5 or higher. From that version on, `useLazyQuery` will return a Promise
